carver.go

```diff
@@ -135,8 +135,6 @@ func (c *Carver) FindLowestEnergySeams() []Seam {
 	// check the immediate three top pixel seam level and
 	// add add the one which has the lowest cumulative energy.
 	for y := c.Height - 2; y >= 0; y-- {
-		left, right = math.MaxFloat64, math.MaxFloat64
```

Comment:
Why this has been eliminated?

Reply:
If you see above, you set this variables, and this is not needed

Reply:
![26149846-ef60-4780-9c98-c1b1b507fe84](https://user-images.githubusercontent.com/888650/36137779-c61e84ae-10a7-11e8-95a6-533199e2df46.png)

Reply:
In contrary, those are needed, because otherwise we always get the minimum zero value, which we try to avoid, since the algorithm start with maximum possible value (which is `MaxFloat64`) then on each iteration will get the smallest value between the two. So there is concrete reason for those assignments.

Reply:
Yep, for left it is really needed, but for “right” and “middle”, nope..
Because it is do nothing..

I’ll fix code in few hours.. from mobile it’s hard to do..

Reply:
It's needed for left and right too. The algorithm works as it should to work.

Reply:
You set right above..

Reply:
That’s why I remove it

Reply:
Oh.. re-read code.. and see what you talk about.. i’ll try when get to computer